testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeHudi.java

```diff
@@ -0,0 +1,146 @@
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.HADOOP;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.common.Minio.MINIO_CONTAINER_NAME;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TEMPTO_PROFILE_CONFIG;
+import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_TRINO_ETC;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+/**
+ * EnvSinglenodeHudi test environment consists of:
+ * - Hive (used for metastore) (HDP 3.1)
+ * - Spark with Hudi
+ * - MinIO S3-compatible storage to store table data
+ */
+@TestsEnvironment
+public class EnvSinglenodeHudi
+        extends EnvironmentProvider
+{
+    private static final int SPARK_THRIFT_PORT = 10213;
+
+    private static final String SPARK_CONTAINER_NAME = "spark";
+
+    private static final String DEFAULT_S3_BUCKET_NAME = "trino-ci-test";
+
+    private final DockerFiles dockerFiles;
+    private final PortBinder portBinder;
+    private final String hadoopImagesVersion;
+    private final DockerFiles.ResourceProvider configDir;
+
+    @Inject
+    public EnvSinglenodeHudi(
+            Standard standard,
+            Hadoop hadoop,
+            DockerFiles dockerFiles,
+            EnvironmentConfig config,
+            PortBinder portBinder,
+            Minio minio)
+    {
+        super(ImmutableList.of(standard, hadoop, minio));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+        this.hadoopImagesVersion = config.getHadoopImagesVersion();
+        this.configDir = dockerFiles.getDockerFilesHostDirectory("conf/environment/singlenode-hudi");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        String s3Bucket = getS3Bucket();
+
+        // Using hdp3.1 so we are using Hive metastore with version close to versions of  hive-*.jars Spark uses
+        builder.configureContainer(HADOOP, container -> {
+            container.setDockerImageName("ghcr.io/trinodb/testing/hdp3.1-hive:" + hadoopImagesVersion);
+        });
+
+        builder.addConnector("hive", forHostPath(configDir.getPath("hive.properties")));
+        builder.addConnector(
+                "hudi",
+                forHostPath(configDir.getPath("hudi.properties")),
+                CONTAINER_TRINO_ETC + "/catalog/hudi.properties");
+
+        builder.configureContainer(TESTS, dockerContainer -> {
+            dockerContainer.withEnv("S3_BUCKET", s3Bucket)
+                    .withCopyFileToContainer(
+                            forHostPath(dockerFiles.getDockerFilesHostPath("conf/tempto/tempto-configuration-for-hive3.yaml")),
+                            CONTAINER_TEMPTO_PROFILE_CONFIG);
```

Comment:
Can we avoid using a deprecated constant?

Reply:
The suggested method `EnvironmentContainers#configureTempto` to replace this deprecated constant is not generic enough. It assumes that `tempto-configuration.yaml` is config resource name. I can do this in a follow up PR.